diff -durN dgen-sdl-1.28.orig/mz80/makez80.c dgen-sdl-1.28/mz80/makez80.c
--- dgen-sdl-1.28.orig/mz80/makez80.c	2011-11-17 22:49:02.000000000 +0100
+++ dgen-sdl-1.28/mz80/makez80.c	2011-12-11 18:18:39.121410978 +0100
@@ -7121,6 +7121,8 @@
 {
 	if (MZ80_ASSEMBLY_X86 == bWhat)
 	{
+		fprintf(fp, "\n");
+		fprintf(fp, "section .note.GNU-stack noalloc noexec nowrite progbits\n");
 	}
 	else
 	if (MZ80_C == bWhat)
diff -durN dgen-sdl-1.28.orig/star/star.c dgen-sdl-1.28/star/star.c
--- dgen-sdl-1.28.orig/star/star.c	2011-11-17 22:49:02.000000000 +0100
+++ dgen-sdl-1.28/star/star.c	2011-12-11 18:13:49.853913096 +0100
@@ -2428,6 +2428,8 @@
 ** the jump table / loop info table
 */
 static void suffixes(void) {
+	emit("\n");
+	emit("section .note.GNU-stack noalloc noexec nowrite progbits\n");
 }
 
 /****************************************************************************
diff -durN dgen-sdl-1.28.orig/x86_ctv.asm dgen-sdl-1.28/x86_ctv.asm
--- dgen-sdl-1.28.orig/x86_ctv.asm	2011-11-17 22:49:02.000000000 +0100
+++ dgen-sdl-1.28/x86_ctv.asm	2011-12-11 18:09:15.354529188 +0100
@@ -162,3 +162,6 @@
 pop  edi
 xor eax,eax
 ret
+
+section .note.GNU-stack noalloc noexec nowrite progbits
+
diff -durN dgen-sdl-1.28.orig/x86_memcpy.asm dgen-sdl-1.28/x86_memcpy.asm
--- dgen-sdl-1.28.orig/x86_memcpy.asm	2011-11-17 22:49:02.000000000 +0100
+++ dgen-sdl-1.28/x86_memcpy.asm	2011-12-11 18:09:25.474623101 +0100
@@ -33,3 +33,6 @@
 ret
 
 ; --------------------------------------
+
+section .note.GNU-stack noalloc noexec nowrite progbits
+
diff -durN dgen-sdl-1.28.orig/x86_mmx_memcpy.asm dgen-sdl-1.28/x86_mmx_memcpy.asm
--- dgen-sdl-1.28.orig/x86_mmx_memcpy.asm	2011-11-17 22:49:02.000000000 +0100
+++ dgen-sdl-1.28/x86_mmx_memcpy.asm	2011-12-11 18:09:40.598763991 +0100
@@ -40,3 +40,6 @@
 ret
 
 ; --------------------------------------
+
+section .note.GNU-stack noalloc noexec nowrite progbits
+
diff -durN dgen-sdl-1.28.orig/x86_tiles.asm dgen-sdl-1.28/x86_tiles.asm
--- dgen-sdl-1.28.orig/x86_tiles.asm	2011-11-17 22:49:02.000000000 +0100
+++ dgen-sdl-1.28/x86_tiles.asm	2011-12-11 18:10:12.001055826 +0100
@@ -1088,3 +1088,6 @@
 	__highpal	dd 0
 
 ; ----------------------------------------
+
+section .note.GNU-stack noalloc noexec nowrite progbits
+
