--- dgen-1.23.orig/sdl/sdl.cpp
+++ dgen-1.23/sdl/sdl.cpp
@@ -157,13 +157,15 @@
   );
 }
 
+void pd_initoptions()
+{
+	fullscreen = dgen_fullscreen;
+	x_scale = y_scale = dgen_scale;
+}
+
 // Handle the switches
 void pd_option(char c, const char *)
 {
-  // Set stuff up from the rcfile first, so we can override it with commandline
-  // options
-  fullscreen = dgen_fullscreen;
-  x_scale = y_scale = dgen_scale;
 #ifdef SDL_OPENGL_SUPPORT
   opengl = dgen_opengl;
   if(opengl)

--- dgen-1.23.orig/main.cpp
+++ dgen-1.23/main.cpp
@@ -236,6 +236,8 @@
   // Parse the RC file
   parse_rc(NULL);
 
+  pd_initoptions();
+  
   // Check all our options
   strcpy(temp, "s:hvr:n:p:RPjd:D:");
   strcat(temp, pd_options);
--- dgen-1.23.orig/pd.h
+++ dgen-1.23/pd.h
@@ -52,6 +52,8 @@
 // This should be a list of all the command-line options specific to this
 // platform, in the form given to getopt(3), i.e "a:b::c".
 extern char *pd_options;
+// This is called once, during startup, before pd_option is.
+void pd_initoptions();
 // And, this is called to handle platform-specific stuff.
 void pd_option(char c, const char *optarg);
 
